mynav.h

```diff
@@ -0,0 +1,101 @@
+/*
+    Handle MyNav TRC format .trc and .ftn files
+
+    Copyright (c) 2014-2020 Ralf Horstmann <ralf@ackstorm.de>
+    Copyright (C) 2014-2020 Robert Lipe, robertlipe+source@gpsbabel.org
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+ */
+
+#ifndef MYNAV_H_INCLUDED_
+#define MYNAV_H_INCLUDED_
+
+#include <QtCore/QString>
+#include <QtCore/QVector>
+
+#include "defs.h"
+#include "format.h"
+
+class MyNavFormat : public Format
+{
+public:
+  ff_type get_type() const override
+  {
+    return ff_type_file;
+  }
+
+  QVector<ff_cap> get_cap() const override
+  {
+    return {
+      ff_cap_none,  // waypoints
+      ff_cap_read,  // tracks
+      ff_cap_none   // routes
+    };
+  }
+
+  QString get_encode() const override
+  {
+    return CET_CHARSET_ASCII;
+  }
+
+  int get_fixed_encode() const override
+  {
+    return 0;
+  }
+
+  void rd_init(const QString& fname) override;
+  void read() override;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/manual/gpsbabel/gpsbabel/pullRequest?prid=4885950)